src/interface/ui/activitythread/ActivityThread.cpp

```diff
@@ -985,6 +992,10 @@ auto ActivityThread::validate_account(
 ActivityThread::~ActivityThread()
 {
     wait_for_startup();
-    signal_shutdown().get();
+    try {
+        signal_shutdown().get();
+    } catch (const std::exception&) {
+        // TODO MT-34 improve
```

Comment:
I suggest to print what went wrong before MT-34 will be implemented

Reply:
More is needed, hence the TODO and ticket, but I have added a minimal diagnostic line especially for you :-)